### ISSUE TITLE
Compiling connector on java 11 throws exception `java.lang.UnsupportedOperationException`.

### ISSUE THREAD

**Issue by maasg** [GoogleCloudDataproc/spark-bigquery-connector#340]
As reported in #200 and #290, the project fails to compile against Java 11 and requires Java 8 for compilation.

**Comment by davidrabinowitz** [comment]
As written [here](https://github.com/GoogleCloudDataproc/spark-bigquery-connector/issues/290#issuecomment-749280836), 

> For the moment, please compile the connector with Java 8. Java 11 compilation support is on our roadmap.

**Comment by maasg** [comment]
@davidrabinowitz I opened a new ticket to create a PR against it. Here: https://github.com/GoogleCloudDataproc/spark-bigquery-connector/pull/341
That patch seems to work well.
Is there something more to consider to support Java 11?
We are actually blocked by the old arrow dependency in the project. Using Java 11 provides a low friction upgrade.
